components/core/CMakeLists.txt

```
@@ -288,6 +305,16 @@ if(CLP_NEED_YSTDLIB)
     add_subdirectory("${CLP_YSTDLIB_SOURCE_DIRECTORY}" "${CMAKE_BINARY_DIR}/ystdlib" EXCLUDE_FROM_ALL)
 endif()
 
+# Find and setup ZLIB
+if(CLP_NEED_ZLIB)
+    find_package(ZLIB REQUIRED)
+    if(ZLIB_FOUND)
+        message(STATUS "Found ZLIB ${ZLIB_VERSION_STRING}")
+    else()
+        message(FATAL_ERROR "Could not find ${CLP_LIBS_STRING} libraries for ZLIB")
+    endif()
```

I know it is following existing code, but imo we should try to stop this practice. `REQUIRED` will exit with failure if the package isn't found, so the if statement will always be true and that else block is unreachable.
```suggestion
    message(STATUS "Found ZLIB ${ZLIB_VERSION_STRING}")
```

For reference see: https://cmake.org/cmake/help/latest/command/find_package.html#basic-signature